### DISCUSSION TITLE
How read multiple pages excel file.

### DISCUSSION THREAD

**Discussion by StefanoTesla** [spatie/simple-excel#91] [Q&A]
Hello everyone, I'm coming from Laravel Excel and rap2hpoutre/fast-excel.

And now I would like to try olso simple-excel!

I don't understand from the documentation how to read all the pages of the file.

How can I do that??

In the first step I need to get only the header and the page name, I can read the header but I don't know how to get the page name.

Can someone help me?

**Comment by Kristories** [comment]
Did you mean [worksheet](https://www.excel-easy.com/basics/worksheets.html)?

If it what you mean, I'll try to give the answer.

Under the hood this package uses the [box/spout](https://github.com/openspout/openspout) package. You can get to the underlying reader that implements `\OpenSpout\Reader\ReaderInterface` by calling the `getReader` method.

```
$reader = SimpleExcelReader::create($file)->getReader();
```

### Playing with sheets

Retrieve all the sheets :
```
$sheets = $reader->getSheets(); 
```
It is possible to retrieve some sheet's attributes when reading:
```
foreach ($reader->getSheetIterator() as $sheet) {
    $sheetName = $sheet->getName();
}
```
